test/SILGen/availability_overloads.swift

```diff
@@ -1,25 +1,44 @@
-// RUN: %target-swift-frontend -emit-silgen %s
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %target-swift-frontend -emit-module-path %t/availability_overloads_other.swiftmodule -emit-module -primary-file %S/Inputs/availability_overloads_other.swift
+
+// RUN: %target-swift-frontend -swift-version 3 -I %t -emit-silgen -primary-file %s
+// RUN: %target-swift-frontend -swift-version 3 -I %t -emit-silgen -emit-ir %s
+
+// RUN: %target-swift-frontend -swift-version 4 -I %t -emit-silgen -primary-file %s
+// RUN: %target-swift-frontend -swift-version 4 -I %t -emit-silgen -emit-ir %s
+
+// RUN: %target-swift-frontend -swift-version 3 -I %t -emit-module -emit-module-path /dev/null -primary-file %s
+// RUN: %target-swift-frontend -swift-version 4 -I %t -emit-module -emit-module-path /dev/null -primary-file %s
 
 // This is a "don't crash with duplicate definition errors" test.
 // We care about being able to express each of these "redeclarations" when the
 // availability doesn't overlap.
 
-class BeforeAndAfter {
+import availability_overloads_other
+
+// FIXME: Should this work?
+#if false
```

Comment:
@jrose-apple I moved the BeforeAndAfter class into a separate module to make sure we can serialize this stuff properly, too.

Do you think we should be able to override these properties? I guess override checking needs a 'non-overlapping availability' check of some kind too.